CMakeLists.txt

```diff
@@ -208,27 +212,27 @@ install(FILES ${GLADE_FILES} ${ICON_FILES} DESTINATION ${GLADE_FILES_DEST})
 if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
 	configure_file(adi-osc.desktop.cmakein ${CMAKE_CURRENT_BINARY_DIR}/adi-osc.desktop @ONLY)
 	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/adi-osc.desktop
-		DESTINATION ${CMAKE_INSTALL_PREFIX}/share/applications/)
+		DESTINATION ${CMAKE_INSTALL_DATADIR}/applications/)
 
 	# install theme icons
 	install(FILES icons/osc16.png RENAME adi-osc.png
-		DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/16x16/apps)
+		DESTINATION ${CMAKE_INSTALL_DATADIR}/icons/hicolor/16x16/apps)
 	install(FILES icons/osc32.png RENAME adi-osc.png
-		DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/32x32/apps)
+		DESTINATION ${CMAKE_INSTALL_DATADIR}/icons/hicolor/32x32/apps)
 	install(FILES icons/osc64.png RENAME adi-osc.png
-		DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/64x64/apps)
+		DESTINATION ${CMAKE_INSTALL_DATADIR}/icons/hicolor/64x64/apps)
 	install(FILES icons/osc128.png RENAME adi-osc.png
-		DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/128x128/apps)
+		DESTINATION ${CMAKE_INSTALL_DATADIR}/icons/hicolor/128x128/apps)
 	install(FILES icons/osc256.png RENAME adi-osc.png
-		DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/256x256/apps)
+		DESTINATION ${CMAKE_INSTALL_DATADIR}/icons/hicolor/256x256/apps)
 	install(FILES icons/osc.svg RENAME adi-osc.svg
-		DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/scalable/apps)
+		DESTINATION ${CMAKE_INSTALL_DATADIR}/icons/hicolor/scalable/apps)
 
 	# This is a bit "ugly". It makes sure that the hicolor theme directory updates
 	# it's modification date. This makes sure that osc icons are loaded right away.
 	# We can also just remove this and assume the user has do this manually or run
 	# something like `gtk-update-icon-cache -f /usr/share/icons/hicolor`.
-	install(CODE "EXECUTE_PROCESS(COMMAND sh -c \"touch /usr/share/icons/hicolor\")")
+	install(CODE "EXECUTE_PROCESS(COMMAND sh -c \"touch ${CMAKE_INSTALL_DATADIR}/icons/hicolor\")")
```

Comment:
I think here we need to use `CMAKE_INSTALL_FULL_DATADIR`...

Reply:
No, that's fine as it is. Using `CMAKE_INSTALL_FULL_DATADIR` would prevent you to install to the prefix `/usr/local`. The `CMAKE_INSTALL_FULL_*` variants are exactly for the case of polkit, where you want to ignore the install prefix and always install to a fixed location.

Reply:
Unless, of course, icons are of the same category as polkit...

Reply:
Looking at https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html, the XDG specifies that it is searched for under `/usr` and `/usr/local`. So, what to do when somebody uses `/opt` or something completely different as install prefix?

I would like to keep it as it is right now. Now, the icons are installed under the prefix, making them readily accessible for `/usr` and `/usr/local` and need further administrative care for all other prefixes, which is likely the expected behavior. It especially avoids writing to `/usr` when installing to `/opt` and the likes.

Reply:
> No, that's fine as it is. Using CMAKE_INSTALL_FULL_DATADIR would prevent you to install to the prefix /usr/local.

Don't think so. Did you try to run a make install on a normal distro? It will try to use `share/icons/hicolor` to run the touch command which won't exist... When you run `install` with `DESTINATION`, things work because (as specified in cmake [install](https://cmake.org/cmake/help/latest/command/install.html)):

"...If a relative path is given it is interpreted relative to the value of the CMAKE_INSTALL_PREFIX variable..."

> The CMAKE_INSTALL_FULL_* variants are exactly for the case of polkit, where you want to ignore the install prefix 

AFAIU, the FULL_DIR stuff actually works by, typically, prepending the  install prefix... Check [here](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html):

"If the value is not already an absolute path, an absolute path is constructed typically by prepending the value of the CMAKE_INSTALL_PREFIX variable."

> So, what to do when somebody uses /opt or something completely different as install prefix?

I guess in this case, there's no miracles and icons won't work :smile:... But I would say, that's for the user to care.

Reply:
You're right. I don't know why I was sure that it worked for me before.